pkg/service/types.go

```diff
@@ -28,17 +22,42 @@ const (
 	RemoteCluster
 )
 
-// MeshService is the struct defining a service (Kubernetes or otherwise) within a service mesh.
+// MeshService is the struct representing a service (Kubernetes or otherwise) within the service mesh.
 type MeshService struct {
 	// If the service resides on a Kubernetes service, this would be the Kubernetes namespace.
 	Namespace string
 
 	// The name of the service
 	Name string
+
+	// Port is the port number that clients use to access the service.
+	// This can be different than MeshService.TargetPort which represents the actual port number
+	// the application is accepting connections on.
+	// Port maps to ServicePort.Port in k8s: https://pkg.go.dev/k8s.io/api/core/v1#ServicePort
+	Port uint16
+
+	// TargetPort is the port number on which an application accept traffic directed to this MeshService
+	// This can be different than MeshService.Port in k8s.
+	// TargetPort maps to ServicePort.TargetPort in k8s: https://pkg.go.dev/k8s.io/api/core/v1#ServicePort
+	TargetPort uint16
+
+	// Protocol is the protocol served by the service's port
+	Protocol string
```

Comment:
This is great, it prevents all the back and forth code of converting MeshService to k8s service in the code